docs/persistentvolumeclaim.md

```diff
@@ -0,0 +1,149 @@
+
+| PersistentVolumeClaim Parameters | LVM CSI Driver | E2E Coverage |
+| -------------------------------- | -------------- | ------------ |
+| [AccessMode](#accessmode) <br> Supported access modes are <li> ReadWriteOnce </li> </br> | Supported | [Yes](https://github.com/openebs/lvm-localpv/tree/master/e2e-tests/experiments/lvm-localpv-provisioner#readme) |
+| [Storageclass](#storageclassname) | Supported | [Yes](https://github.com/openebs/lvm-localpv/tree/master/e2e-tests/experiments/lvm-localpv-provisioner#readme) |
+| [Capacity Resource](#capacity-resource) | Supported | Yes |
+| [VolumeMode](#volumemode) <br> Supported volume modes are <li> Block </li> <li> Filesystem </li> </br> | Supported | Yes <br> Test exist only for Filesystem mode </br>|
+| [Selectors](#selectors)   | Supported | Pending |
+| [VolumeName](#volumename) | Supported | Pending | 
+
+## PersistentVolumeClaim Parameters
+
+### AccessMode
+
+LVM-LocalPV supports only `ReadWriteOnce` access mode i.e volume can be mounted as read-write by a single node.
+```yaml
+kind: PersistentVolumeClaim
+apiVersion: v1
+metadata:
+  name: csi-lvmpv
+spec:
+  accessModes:
+    - ReadWriteOnce        ## Specify the ReadWriteOnce(RWO) access modes
+  storageClassName: openebs-lvm
+  resources:
+    requests:
+      storage: 4Gi
+```
+
+### StorageClassName
+
+LVM CSI-Driver supports dynamic provision of volume for the PVCs refered to lvm storageclass.
+
+```yaml
+kind: PersistentVolumeClaim
+apiVersion: v1
+metadata:
+  name: csi-lvmpv
+spec:
+  accessModes:
+    - ReadWriteOnce
+  storageClassName: openebs-lvm    ## It must be OpenEBS LVM storageclass
+  resources:
+    requests:
+      storage: 4Gi
+```
+
+### Capacity Resource
+
+Admin/User can specify the desired capacity for lvm volume. Once the CSI-Driver gets request it will check whether available space in underlying volume group. If it has enough space a success respone will be returned to caller else error will be reported.
+
+```yaml
+kind: PersistentVolumeClaim
+apiVersion: v1
+metadata:
+  name: csi-lvmpv
+spec:
+  accessModes:
+    - ReadWriteOnce
+  storageClassName: openebs-lvm
+  resources:
+    requests:
+      storage: 4Gi       ## It must be supported only once
+```
+
+### VolumeMode
+
+LVM-LocalPV supports two kind of volume modes(Defaults to Filesystem):
+- Block  (Block mode can be used in a case where application itself maintains filesystem)
+- Filesystem (Application which requires filesystem as a prerequisite)
+
+```yaml
+kind: PersistentVolumeClaim
+apiVersion: v1
+metadata:
+  name: csi-lvmpv
+spec:
+  accessModes:
+    - ReadWriteOnce
+  storageClassName: openebs-lvm
+  volumeMode: Filesystem     ## Specifies in which mode volume should be attached to pod
+  resources:
+    requests:
+      storage: 4Gi
+```
+
+
+### Selectors
+
+Users can bind the retained lvm volumes to new PersistentVolumeClaim object via selector. Follow below steps
```

Comment:
should we say that if PVs are already labelled?

Reply:
I will update the doc saying if there are no labels exist on PV then update with meaningful labels.